here-naksha-app-service/src/main/java/com/here/naksha/app/service/http/NakshaHttpVerticle.java

```diff
@@ -702,11 +704,18 @@ public void sendRawResponse(
 
   public @NotNull NakshaContext createNakshaContext(final @NotNull RoutingContext routingContext) {
     final NakshaContext ctx = new NakshaContext(AccessLogUtil.getStreamId(routingContext));
-    ctx.setAppId(hubConfig.appId);
     // add streamInfo object to NakshaContext, which will be populated later during pipeline execution
     ctx.attachStreamInfo(AccessLogUtil.getStreamInfo(routingContext));
-    // TODO : Author to be set based on JWT token.
-    // ctx.setAuthor();
+    // extract the JWT from authorization header
+    final JWTPayload jwtPayload = getOrCreateJWT(routingContext);
+    if (jwtPayload == null) {
+      log.error("Cannot detect JWT payload in routing context: {}", routingContext);
+    } else {
+      // attach authorization info into context
+      ctx.setAppId(jwtPayload.appId);
```

Comment:
pls validate (if not already) that the Wikvaya jwt contains appId and userId. AppId is mandatory. so, if wikvaya is not passing this, then after merging the PR and cloud deployment, the Naksha APIs will start failing in cloud env.

Reply:
<img width="969" alt="image" src="https://github.com/xeus2001/xyz-hub/assets/39376416/5e5d8c0d-0b30-4f29-b48b-232850227537">